src/WebJobs.Script/Description/Workers/WorkerFunctionInvoker.cs

```diff
@@ -85,15 +87,20 @@ protected override async Task<object> InvokeCore(object[] parameters, FunctionIn
             return result.Return;
         }
 
-        private async Task DelayUntilFunctionDispatcherInitialized()
+        private async Task DelayUntilFunctionDispatcherInitializedOrShutdown()
         {
             if (_functionDispatcher != null && _functionDispatcher.State == FunctionInvocationDispatcherState.Initializing)
             {
                 _logger.LogDebug($"functionDispatcher state: {_functionDispatcher.State}");
-                await Utility.DelayAsync(WorkerConstants.ProcessStartTimeoutSeconds, WorkerConstants.WorkerReadyCheckPollingIntervalMilliseconds, () =>
+                await Utility.DelayAsync(_functionDispatcher.ErrorEventsThreshold * WorkerConstants.ProcessStartTimeoutSeconds, WorkerConstants.WorkerReadyCheckPollingIntervalMilliseconds, () =>
```

Comment:
Update DelayAsync method to return `Task<bool>` to avoid checking for state again. If the result is false, shutdown

Reply:
You meant to say if the result is true (meaning dispatcher state is still != initialized), shutdown, right?

Reply:
correct. If the result is still not initialized then shutdown